src/compose/ComposeMenu.js

```diff
@@ -63,72 +69,79 @@ export const chooseUploadImageFilename = (uri: string, fileName: ?string): strin
    * automatically converted to JPEG format... but the `fileName` property in
    * the react-native-image-picker response still has the `.HEIC` extension.
    */
+  // TODO: Is it still true, after the react-native-image-picker upgrade,
+  // that `fileName` can end in .HEIC?
   if (/\.jpe?g$/i.test(uri)) {
-    return name.replace(/\.heic$/i, '.jpeg');
+    const result = nameWithoutPrefix.replace(/\.heic$/i, '.jpeg');
+    if (result !== nameWithoutPrefix) {
+      logging.info('OK, so .HEIC to .jpeg replacement still seems like a good idea.');
+    }
+    return result;
   }
 
-  return name;
+  return nameWithoutPrefix;
 };
 
 class ComposeMenuInner extends PureComponent<Props> {
-  uploadFile = (uri: string, fileName: ?string) => {
+  handleImagePickerResponse = response => {
     const { dispatch, destinationNarrow } = this.props;
-    dispatch(uploadFile(destinationNarrow, uri, chooseUploadImageFilename(uri, fileName)));
-  };
 
-  handleImagePickerResponse = (
-    response: $ReadOnly<{
-      didCancel: boolean,
-      // Upstream docs are vague:
-      // https://github.com/react-native-community/react-native-image-picker/blob/master/docs/Reference.md
-      error?: string | void | null | false,
-      uri: string,
-      // Upstream docs are wrong (fileName may indeed be null, at least on iOS);
-      // surfaced in https://github.com/zulip/zulip-mobile/issues/3813:
-      // https://github.com/react-native-community/react-native-image-picker/issues/1271
-      fileName: ?string,
-      ...
-    }>,
-  ) => {
-    if (response.didCancel) {
+    if (response.didCancel === true) {
+      return;
+    }
+
+    const errorCode: string | void = response.errorCode;
+    if (errorCode != null) {
+      showErrorAlert('Error', response.errorMessage);
       return;
     }
 
-    // $FlowFixMe[sketchy-null-string]
-    /* $FlowFixMe[sketchy-null-bool]
-       Upstream API is unclear. */
-    const error: string | null = response.error || null;
-    if (error !== null) {
-      showErrorAlert('Error', error);
+    // TODO: support sending multiple files; see library's docs for how to
+    // let `assets` have more than one item in `response`.
+    const firstAsset = response.assets && response.assets[0];
+
+    const { uri, fileName } = firstAsset ?? {};
+
+    if (!firstAsset || uri == null || fileName == null) {
+      // TODO: If we need to keep this, wire up the user-facing string for
+      // translation.
+      showErrorAlert('Error', 'Something went wrong, and your message was not sent.');
+      logging.error('Unexpected response from image picker', {
+        firstAssetUndefined: firstAsset === undefined,
+        uriUndefined: uri === undefined,
+        fileNameUndefined: fileName === undefined,
```

Comment:
Probably these three diagnostic booleans should match the three `||` operands above.